Realm/Realm/Realm.cs

```diff
@@ -368,7 +368,7 @@ public override int GetHashCode()
         {
             ThrowIfDisposed();
 
-            return (int)SharedRealmHandle.DangerousGetHandle();
+            return (int)((long)SharedRealmHandle.DangerousGetHandle() % int.MaxValue);
```

Comment:
`DangerousGetHandle()` returns `IntPtr` which is long on 64 bit devices. I'm not sure how it has worked until now, but on UWP it consistently throws out of range exception. There are different strategies for wrapping it in an int - I chose one that seemed fine, but I'm open to suggestions if anyone has a better idea :)